cve_bin_tool/input_engine.py

```diff
@@ -115,24 +115,30 @@ def strip_remark(detail) -> str:
                         ref = affect["ref"]
                         version = None
                         vendor = None
+                        # Expected ref format is urn:cbt:{bom_version}/{vendor}#{product}-{version}
                         if "#" in ref:
                             # Extract product information after # delimiter
                             p = ref.split("#")[1]
                             # Last element is version, rest is product which may include -
                             elements = p.rsplit("-", 1)
                             product = elements[0]
                             version = elements[1]
-                            # Now find vendor
-                            vendor_package_pair = self.cvedb.get_vendor_product_pairs(
-                                product
-                            )
-                            if vendor_package_pair != []:
-                                # FIXME https://github.com/intel/cve-bin-tool/issues/2320
-                                vendor = vendor_package_pair[
-                                    len(vendor_package_pair) - 1
-                                ]["vendor"]
+                            # Note that cbt (cve-bin-tool) and cdx (cyclonedx)
+                            # namespaces are different
+
+                            vendor_match = None
+                            if re.match("urn:cbt", ref):
+                                # cbt looks like "urn:cbt:1/gnu#glibc-2.33"
+                                # vendor is between / and #
+                                vendor_match = re.match("urn:cbt:[^/]*/([^#]*)#", ref)
+                            # cdx looks like "urn:cdx:NOTKNOWN/1#libjpeg-turbo-2.0.1"
```

Comment:
The alternate thing we could do here is match the NOTKNOWN but it doesn't seem like this is intended to be vendor so I just didn't match it for now.